newdle/client/src/components/Timeline.js

```diff
@@ -233,19 +234,56 @@ function TimelineContent({busySlots, minHour, maxHour}) {
 
 TimelineContent.propTypes = {
   busySlots: PropTypes.array.isRequired,
-  minHour: PropTypes.number,
-  maxHour: PropTypes.number,
-};
-
-TimelineContent.defaultProps = {
-  minHour: 6,
-  maxHour: 24,
+  minHour: PropTypes.number.isRequired,
+  maxHour: PropTypes.number.isRequired,
 };
 
-export default function Timeline({date, availability, minHour, maxHour, hourStep}) {
+export default function Timeline({date, availability, defaultMinHour, defaultMaxHour, hourStep}) {
+  const [minHour, setMinHour] = useState(defaultMinHour);
+  const [maxHour, setMaxHour] = useState(defaultMaxHour);
+  const candidates = useSelector(getTimeslotsForActiveDate);
+  const duration = useSelector(getDuration);
   const hourSeries = _.range(minHour, maxHour + hourStep, hourStep);
   const hourSpan = maxHour - minHour;
+  const defaultHourSpan = defaultMaxHour - defaultMinHour;
   const busySlots = calculateBusyPositions(availability, minHour, maxHour);
+
+  const computeHourSpan = useCallback(() => {
```

Comment:
I don't think `useCallback` is needed here - you are not passing the function to anything so there is no benefit in preserving its identity between renders.

Reply:
ESLint complains then 
> The 'computeHourSpan' function makes the dependencies of useEffect Hook (at line 285) change on every render. Move it inside the useEffect callback. Alternatively, wrap the 'computeHourSpan' definition into its own useCallback() Hook.

Reply:
ah it's a dep of useEffect. then you do need the useCallback